### PR TITLE
fix(mme): malformed packet(Authentication request) in scale test

### PR DESCRIPTION
**Pull request by Lkishor123**
Signed-off-by: LKishor123 <laawanya.kishor@wavelabs.ai>

## Summary
Malformed packet(Authentication request) in scale test.
 1. Changed amf_ue_ngap_id from uint32_t to uint64_t

## Test Plan 
 1. UERANSIM:
     The test is passing till amf-ue-ngap-id : 4294967293 but amf-ue-ngap-id : 4294967297 is failing, screenshots for the same are attached.
amf_ue_ngap_id_v1
amf_ue_ngap_id_v2

 2. TeraVM

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by ulaskozat — CHANGES_REQUESTED** [review]
Please take care of github action warnings

**Comment by Lkishor123** [comment]
> Please take care of github action warnings

Most warnings are addressed, some cpplint error are ignored for now as they will be resolved in other PR's.

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c [hunk 1]
```diff
@@ -862,7 +835,7 @@ void ngap_handle_conn_est_cnf(
       if (er.encoded <= 0) {
         OAILOG_ERROR(
             LOG_NGAP, "PDU Session Resource Request IE encode error \n");
-        OAILOG_FUNC_RETURN(LOG_NGAP, RETURNerror);
+        return;
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c]: use `OAILOG_FUNC_OUT(LOG_NGAP);`

Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c]: Comment addressed.

**Comment by ulaskozat** [comment]
Fix merge conflicts

File: lte/gateway/c/core/oai/tasks/amf/amf_app_timer_management.cpp [hunk 2]
```diff
@@ -52,7 +51,7 @@ int AmfUeContext::StartTimer(
   int timer_id = -1;
   if ((timer_id = start_timer(
            &amf_app_task_zmq_ctx, msec, repeat, handler, nullptr)) != -1) {
-    amf_app_timers.insert(std::pair<int, uint32_t>(timer_id, arg));
+    amf_app_timers.insert(std::pair<int, timer_arg_t>(timer_id, arg));
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_timer_management.cpp:55]: legitimate github actions comment.

Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_timer_management.cpp:55]: Comment addressed.

File: lte/gateway/c/core/oai/tasks/amf/amf_authentication.cpp [hunk 3]
```diff
@@ -832,7 +832,9 @@ int amf_send_authentication_request(
 
     if (rc != RETURNerror) {
       OAILOG_WARNING(
-          LOG_NAS_AMF, " T3560:Start Authenthication Timer for ue id:%d\n",
+          LOG_NAS_AMF,
+          " T3560:Start Authenthication Timer for "
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/amf/amf_authentication.cpp]: typo: `Authenthication` -> `Authentication`

Comment (ulaskozat) [reply lte/gateway/c/core/oai/tasks/amf/amf_authentication.cpp]: leave space after `:`

File: lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp [hunk 4]
```diff
@@ -110,7 +110,8 @@ int amf_proc_deregistration_request(
   OAILOG_FUNC_IN(LOG_NAS_AMF);
   OAILOG_DEBUG(
       LOG_NAS_AMF,
-      "Processing deregistration UE-id = %d "
+      "Processing deregistration UE-id = " AMF_UE_NGAP_ID_FMT
+      " "
       "type = %d\n",
       ue_id, params->de_reg_type);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: looks like too many lines to me?

File: lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp [hunk 5]
```diff
@@ -125,7 +126,8 @@ int amf_proc_deregistration_request(
   if (!amf_ctx) {
     OAILOG_DEBUG(
         LOG_NAS_AMF,
-        "AMF icontext not present for UE-id = %d "
+        "AMF icontext not present for UE-id = " AMF_UE_NGAP_ID_FMT
+        " "
         "type = %d\n",
         ue_id, params->de_reg_type);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: looks too many lines.

File: lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp [hunk 6]
```diff
@@ -265,7 +271,10 @@ void amf_remove_ue_context(ue_m5gmm_context_s* ue_context_p) {
 
   if (found_ue_id != ue_context_map.end()) {
     OAILOG_DEBUG(
-        LOG_AMF_APP, "Removed ue id = %u entry from the ue context map\n",
+        LOG_AMF_APP,
+        "Removed ue id = " AMF_UE_NGAP_ID_FMT
+        " entry from the "
+        "ue context map\n",
         ue_context_p->amf_ue_ngap_id);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: looks too many lines, is this formatted by linter?

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c [hunk 7]
```diff
@@ -708,14 +708,13 @@ status_code_e ngap_amf_handle_initial_context_setup_response(
...
+    asn_INTEGER2ulong(
+        &ie->value.choice.AMF_UE_NGAP_ID, (uint64_t*) &amf_ue_ngap_id);
+    if ((ue_ref_p = ngap_state_get_ue_amfid(amf_ue_ngap_id)) == NULL) {
       OAILOG_DEBUG(
           LOG_NGAP,
-          "No UE is attached to this amf UE ngap id: " AMF_UE_NGAP_ID_FMT
-          " %u(10)\n",
-          (uint32_t) amf_ue_ngap_id, (uint32_t) amf_ue_ngap_id);
+          "No UE is attached to this amf UE ngap id: " AMF_UE_NGAP_ID_FMT "\n",
+          amf_ue_ngap_id);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c]: can be squeezed in fewer lines.

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c [hunk 8]
```diff
@@ -1235,7 +1238,7 @@ status_code_e ngap_amf_handle_ue_context_release_complete(
         LOG_NGAP,
         " UE Context Release commplete: Ng context cleared. Ignore message for "
         "ueid " AMF_UE_NGAP_ID_FMT "\n",
-        (uint32_t) amf_ue_ngap_id);
+        amf_ue_ngap_id);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c]: too many lines.

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c [hunk 9]
```diff
@@ -1583,7 +1587,9 @@ void ngap_amf_handle_ue_context_rel_comp_timer_expiry(
       (const hash_key_t) ue_ref_p->amf_ue_ngap_id, &imsi64);
 
   OAILOG_DEBUG_UE(
-      LOG_NGAP, imsi64, "Expired- UE Context Release Timer for UE id  %d \n",
+      LOG_NGAP, imsi64,
+      "Expired- UE Context Release Timer"
+      " for UE id " AMF_UE_NGAP_ID_FMT "\n",
       ue_ref_p->amf_ue_ngap_id);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c]: too many lines

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c [hunk 10]
```diff
@@ -1620,7 +1626,8 @@ void ngap_amf_release_ue_context(
   OAILOG_FUNC_IN(LOG_NGAP);
   DevAssert(ue_ref_p != NULL);
   OAILOG_DEBUG_UE(
-      LOG_NGAP, imsi64, "Releasing UE Context for UE id  %d \n",
+      LOG_NGAP, imsi64,
+      "Releasing UE Context for UE id" AMF_UE_NGAP_ID_FMT "\n",
       ue_ref_p->amf_ue_ngap_id);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c]: too many lines

File: lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp (hunk 6 above)
Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: Comment addressed, I think it was auto formatted.

File: lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp (hunk 5 above)
Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: Comment addressed.

File: lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp (hunk 4 above)
Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: Comment addressed.

File: lte/gateway/c/core/oai/tasks/amf/amf_authentication.cpp (hunk 3 above)
Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/amf/amf_authentication.cpp]: Comment addressed.

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c (hunk 7 above)
Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c]: Comment addressed.

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c (hunk 8 above)
Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c]: Comment addressed.

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c (hunk 9 above)
Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c]: Comment addressed.

File: lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c (hunk 10 above)
Comment (Lkishor123) [reply lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c]: Comment addressed.

**Review by ulaskozat — APPROVED** [review]
LGTM

**Comment by rdefosse** [comment]
@ulaskozat @Lkishor123 @Scott8440 @themarwhal @hcgatewood @pshelar @panyogesh @sureshgorijavolu 

I just merged 1 pull request that reduces the number of compilation warnings and we are back to **40** new ones!

`https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/6390/artifact/build_results_magma_oai_mme.html`

`https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/6352/artifact/build_results_magma_oai_mme.html`

Can we make an effort on that topic!

I am tired of doing `code cleanup` duties. I won't do it this time!

Regards
Raphael

**Comment by ulaskozat** [comment]
@sureshgorijavolu and @panyogesh can you make a sweep and fix all the compilation warnings on the 5G code.

@rdefosse are the compilation warnings specific to the changes in this PR?

**Pull request merged**